File: src/Placeholder.Models/StubModel.cs
using YamlDotNet.Serialization;

namespace Placeholder.Models
{
   public class StubModel
   {
      [YamlMember(Alias = "id")]
      public string Id { get; set; }

      [YamlMember(Alias = "conditions")]
      public StubConditionsModel Conditions { get; set; }

      [YamlMember(Alias = "response")]
      public StubResponseModel Response { get; set; }
   }
}

File: gui2/src/domain/stub/stub-model.ts
import type { StubConditionsModel } from "@/domain/stub/stub-conditions-model";
import type { StubResponseModel } from "@/domain/stub/stub-response-model";

export interface StubModel {
  id: string;
  conditions: StubConditionsModel;
  response: StubResponseModel;
  priority: number;
  tenant?: string;
  description?: string;
  enabled: boolean;
  scenario?: string;
}
